### ISSUE TITLE
Main tldr.sh site taken by a domain squatter

### ISSUE THREAD

**Issue by coandco-2fa** [tldr-pages/tldr#2253]
http://tldr.sh currently points to a generic domain-squatting page, and the tldr datafiles don't appear to be hosted anywhere else.  Is this a known issue?

**Comment by agnivade** [comment]
Thanks. Did not know that. tldr.sh is owned by @ostera

**Comment by agnivade** [comment]
I have removed the CNAME from tldr-pages.github.io. Please give some time for DNS changes to propagate.

**Comment by agnivade** [comment]
Looks like it is propagated now. Let us keep this open until we can fix the domain.

**Comment by ostera** [comment]
Hello! I've resolved that the domain renewal didn't happen (a.k.a. I missed the emails) and it should be up and running now.

Sorry for any inconvenience it may have caused 🙌🏼

**Comment by agnivade** [comment]
@ostera - Can more than one person have access to the dns account ? It would be good to have more than one eye on it.

**Comment by mflint** [comment]
So what's the recommended URL for clients to use? My iOS app is currently broken, but I just wanna do the right thing.

**Comment by MasterOdin** [comment]
By default, both the official node and python apps download the entire repo from when it's requested (https://tldr-pages.github.io/assets/tldr.zip) and the python app will also use the raw github links when downloading individual pages. Given that the tldr.sh just CNAME to github, it's probably recommended to just use the github links themselves.

**Comment by agnivade** [comment]
The recommended url is - https://tldr-pages.github.io/assets/tldr.zip. But I have brought back the CNAME just now. Things should be back to normal in a few hours.

**Comment by mflint** [comment]
Follow-up question for @agnivade: `https://tldr-pages.github.io/assets/tldr.zip` gives me a 301 redirect to `http://tldr.sh/assets/tldr.zip` which, quite correctly, iOS refuses to honour.

Are there any plans to host the zip using https? (Or will I need to add a dirty http exception into my iOS app?)

**Comment by agnivade** [comment]
Hmm .. that is strange. I will have to look into github documentation. github pages just allow a CNAME to be specified. So theoretically, it should use the same protocol, given that `https://tldr.sh/assets/tldr.zip` works fine.

Do you have any off-hand knowledge on this ?

**Comment by sbrl** [comment]
Enforcing HTTPS sounds like a good thing - and I suspect that it might fix that issue, @mflint. I've just checked, and it looks like there's a bit of a problem:

![selection_055](https://user-images.githubusercontent.com/9929737/44632645-1d714e00-a976-11e8-8ac5-3b29f6029648.png)

Upon investigation, GitHub's official [help page](https://help.github.com/articles/troubleshooting-custom-domains/#https-errors) says that if we're using an `A` record, we should point at one of these IPs:

 - 185.199.108.153
 - 185.199.109.153
 - 185.199.110.153
 - 185.199.111.153

....but doing a `dig tldr.sh` reveals the following:

```dns
tldr.sh.		181	IN	A	104.18.48.121
tldr.sh.		181	IN	A	104.18.49.121
```

Is there any reason why we can't use a `CNAME` instead? As I don't currently have access to the DNS record settings, I can't alter the DNS records to fix the issue. I believe @ostera to have access. Perhaps you could take a look please @ostera? (and is there any chance of another maintainer/owner of tldr-pages getting access to the DNS account to have another pair of eyes on it, just in case please?)

**Comment by mflint** [comment]
Disclaimer: I'm no DNS expert.

But: you may not need to use `CNAME`. My (GitHub-pages, SSL) domain reports this:

    {mysite}.net.	3600	IN	A	185.199.111.153
    {mysite}.net.	3600	IN	A	185.199.109.153
    {mysite}.net.	3600	IN	A	185.199.110.153
    {mysite}.net.	3600	IN	A	185.199.108.153

Looks like the `185.199.*.153` IPs are GitHub; your `104.18.*.121` IPs are a Cloudflare front. This is getting a little beyond my area of expertise ;-)

**Comment by sbrl** [comment]
Yeah, maybe not. I think a `CNAME` would update automagically though?

**Comment by agnivade** [comment]
You cannot set CNAME for root domains. It looks like we need to point to the 185.* series of addresses and then remove and add the CNAME file to trigger the process of enabling HTTPS.

https://help.github.com/articles/setting-up-an-apex-domain/

@ostera  - Can we do that ?

**Comment by mflint** [comment]
> You cannot set CNAME for root domains.

True. But some DNS operators have workarounds for it ([ZoneEdit, for example](https://support.zoneedit.com/Knowledgebase/Article/View/1/1/root-domain-aliases-root-aliases-zone-apex-aliases-implemeneted-as-anames)). Again, that's what I'm using for my personal blog.

This is sounding like a non-trivial change - so absolutely _no pressure_ from me to do it :)

**Comment by agnivade** [comment]
I don't think setting CNAME or ALIAS is required. I have A record for my own personal blog, and it works perfectly fine. 

I just did the method I mentioned in my previous comment and confirmed that it works. We just need to point to all the 4 IP addresses and do a remove and add of the CNAME file to trigger the https cert fetch.

**Comment by mflint** [comment]
> We just need to point to all the 4 IP addresses

Maybe - but there's probably a good reason why @ostera chose to use Cloudflare.

**Comment by agnivade** [comment]
Ah, so its pointing to cloudflare. So I think we need to set ssl settings to flexible-ssl and redirect all users to https using a page rule.

**Comment by ostera** [comment]
The Apex record is set to Github Pages IPs

<img width="269" alt="screen shot 2018-08-27 at 12 21 08" src="https://user-images.githubusercontent.com/854222/44654999-eb063600-a9f3-11e8-98bb-2fbc319f1a6b.png">

Which I took from a prior version of Github Pages documentation — it's trivial to update to the news ones. Let me know if you want to try this out today, @agnivade

**Comment by ostera** [comment]
<img width="992" alt="screen shot 2018-08-27 at 12 26 47" src="https://user-images.githubusercontent.com/854222/44655170-7f709880-a9f4-11e8-8d17-b39ff9d6da27.png">

Full config for transparency.

**Comment by ostera** [comment]
Also, I'm happy to configure the domain to someone else's DNS provider 🙌

**Comment by agnivade** [comment]
Let's change them to - 185.199.{108-111}.153. i.e. 4 IPs having 108,109,110,111 in the middle.

**Comment by ostera** [comment]
Wondering if would've taken more time to write 

```
185.199.108.153
185.199.109.153
185.199.110.153
185.199.111.153
```

Than to explain the expansion 😝 XKCD comes to mind: https://xkcd.com/1205/

IPs changed! 😄

**Comment by agnivade** [comment]
Deleted CNAME file. Will revert after a while. Stand by.

**Comment by agnivade** [comment]
It did not work. I see that https://tldr.sh is served by a cloudflare cert. Can we set a redirect to https rule in cloudflare page settings ?

**Comment by ostera** [comment]
@agnivade I went in on CloudFlare and apparently I did have registered the domain there, and that's where the Name Servers are set up in Name.com.

Sent you an invite so you can handle the whole thing from there 🙌

**Comment by agnivade** [comment]
Problem solved. Setup an http to https redirect.

@ostera - I also want to enable HSTS. The only issue with that is that once setup, we cannot disable https anymore. I don't see any issue with it. Do you ?

**Comment by ostera** [comment]
Not from my end. Go bananas! 🐒

**Comment by agnivade** [comment]
Done. Thanks !

**Comment by sbrl** [comment]
Awesome, thanks @ostera and @agnivade :smiley_cat:

**Comment by mflint** [comment]
It's still 301 to http. (I'll keep using https://tldr.sh/assets/tldr.zip for now...)

```
$ curl -v https://tldr-pages.github.io/assets/tldr.zip
{snip}
< HTTP/2 301 
< server: GitHub.com
< content-type: text/html
< location: http://tldr.sh/assets/tldr.zip
```

**Comment by agnivade** [comment]
Try adding an `-L` and see what happens. :)

**Comment by mflint** [comment]
Hmm, try adding `--proto -http` to explicitly disallow http ;-)

`curl -v -L --proto -http https://tldr-pages.github.io/assets/tldr.zip`

(Not sure how you've set this up, but if there are _any_ requests using http, then some clients will - should - fail)

This is my error from the iOS app:

`Error Domain=NSURLErrorDomain Code=-1022 "The resource could not be loaded because the App Transport Security policy requires the use of a secure connection." UserInfo={NSUnderlyingError=0x604000446210 {Error Domain=kCFErrorDomainCFNetwork Code=-1022 "(null)"}, NSErrorFailingURLStringKey=http://tldr.sh/assets/tldr.zip, NSErrorFailingURLKey=http://tldr.sh/assets/tldr.zip, NSLocalizedDescription=The resource could not be loaded because the App Transport Security policy requires the use of a secure connection.}
`

**Comment by agnivade** [comment]
Oh I see. Well, we do not control tldr-pages.github.io. So I don't think there is much we can do to ask github to redirect to https:// when an https:// url is requested.

Let me know if you find something, am happy to do it.

**Comment by sbrl** [comment]
@agnivade I think that we can enforce HTTPS in the [GitHub Pages settings](https://github.com/tldr-pages/tldr-pages.github.io/settings) as I've mentioned above. However, when I try to do so, I'm still met with this:

![selection_057](https://user-images.githubusercontent.com/9929737/44815775-5d377000-abd8-11e8-8be8-d58bdd7b6182.png)

...I'm confused. I thought we fixed that one?

**Comment by mflint** [comment]
@sbrl  The DNS "A records" are still pointing to Cloudflare (104.18.*.121); GitHub requires those A records to point to GitHub (185.199.*.153).

**Comment by agnivade** [comment]
@sbrl - No we couldn't fix that one. Most probably because we are using cloudflare. What I have done is setup a redirect rule from http://tldr.sh to https://tldr.sh.

I don't know what the solution is if a custom domain is to use cloudflare.

**Comment by MasterOdin** [comment]
Do you really need Cloudflare? My understanding was that it was desirable back when Github pages did not support https for custom domains, but now that it does, you're probably better off dropping cloudflare and then having all redirects be https -> https.

**Comment by sbrl** [comment]
Won't a redirect like that cause a loop https -> http -> https?
